determine the control name dynamically

afeld opened the pull request:
Builds on #4. Closes #11.

This will allow the templater to be run over the full document, so that it works on more than just the hard-coded `AC-2 (1)`.

afeld committed:
create the output directory if it doesnt exist

afeld committed:
create a ControlTable struct

afeld committed:
create a controlName method

afeld committed:
pull the name of the control dynamically

afeld committed:
fix the matching of the table header

Apparently the XPath syntax was incorrect, or isn't supported.

afeld committed:
ensure that only the subtree of the ControlTable is searched

Fixes handling of multiple control tables.

afeld committed:
Code Climate fixes in models file

afeld committed:
fill in the Control Summary Information

...as well as the Control Enhancement Summary.

afeld committed:
simplify xpath

afeld commented:
@jcscottiii I'm going to try and figure out how to do better testing of this, but mind taking a look at what I've got?

jcscottiii commented:
👓 _looking_

jcscottiii commented on templater/models_test.go:
```diff
@@ -0,0 +1,28 @@
+)
+
+var _ = Describe("ControlTable", func() {
+	Describe("Fill", func() {
```
I know you have a fixture at `simplified_table.xml` is there a way of getting that xml data and putting it in here as a string? I know docx is really ugly so the answer maybe no / not worth it.

Only reason I ask would be to try cases to hit all the conditional branches for all the methods here by easily modifying the strings per case.

afeld replied:
> is there a way of getting that xml data and putting it in here as a string?

It would be _huge_, so would prefer to keep it as a separate file. Went with templatizing the file to make it a bit more flexible.

jcscottiii commented on templater/templater.go:
```diff
@@ -48,48 +49,21 @@ func printNodes(nodes []xml.Node) {
 	if err != nil {
 		return
 	}
 	for _, table := range tables {
```
would like to see a test of going through and filling multiple tables in a doc

jcscottiii commented on templater/models.go:
```diff
@@ -0,0 +1,88 @@
+		return
+	}
+
+	regex := regexp.MustCompile(`[A-Z]{2}-\d+( \(.\))?`)
```
comment about the types of valid case(s) this should catch

afeld committed:
match control enhancements > 9

afeld commented:
## TODOs
- [x] Address feedback above
- [x] Look into why the matches seem to be stopping at SI-2 when running on the full SSP

afeld committed:
templatize the test table

afeld committed:
add test for filling in control tables

afeld committed:
add integration test for TemplatizeWordDoc

afeld committed:
remove unused functions

afeld committed:
ensure word documents are closed

afeld committed:
handle control table headings with inconsistent whitespace

The summary tables after SI-2 have an extra space, so weren't getting
picked up.

afeld committed:
move the ControlTable to a models package

afeld commented:
Updated! Still want to figure out how to do better testing, but at least the coverage is _better_ now.
